File: schemaorg-rs/src/geo_circle.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///<https://schema.org/additionalType>
///<https://schema.org/Text>
///<https://schema.org/URL>
pub type GeoCircleAdditionalTypeFieldEnum = String;
///<https://schema.org/GeoCircle>
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct GeoCircle {
    #[serde(rename = "@context")]
    pub context: String,
    ///<https://schema.org/box>
    #[serde(rename = "box")]
    #[serde_as(as = "OneOrMany<_>")]
    pub _box: Vec<String>,
    ///<https://schema.org/disambiguatingDescription>
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///<https://schema.org/circle>
    #[serde(rename = "circle")]
    #[serde_as(as = "OneOrMany<_>")]
    pub circle: Vec<String>,
    ///<https://schema.org/elevation>
    #[serde(rename = "elevation")]
    #[serde_as(as = "OneOrMany<_>")]
    pub elevation: Vec<GeoCircleElevationFieldEnum>,
    ///<https://schema.org/description>
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<GeoCircleDescriptionFieldEnum>,
    ///<https://schema.org/name>
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///<https://schema.org/geoMidpoint>
    #[serde(rename = "geoMidpoint")]
    #[serde_as(as = "OneOrMany<_>")]
    pub geo_midpoint: Vec<GeoCoordinates>,
    ///<https://schema.org/url>
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///<https://schema.org/polygon>
    #[serde(rename = "polygon")]
    #[serde_as(as = "OneOrMany<_>")]
    pub polygon: Vec<String>,
    ///<https://schema.org/subjectOf>
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<GeoCircleSubjectOfFieldEnum>,
    ///<https://schema.org/alternateName>
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///<https://schema.org/postalCode>
    #[serde(rename = "postalCode")]
    #[serde_as(as = "OneOrMany<_>")]
    pub postal_code: Vec<String>,
    ///<https://schema.org/potentialAction>
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///<https://schema.org/line>
    #[serde(rename = "line")]
    #[serde_as(as = "OneOrMany<_>")]
    pub line: Vec<String>,
    ///<https://schema.org/geoRadius>
    #[serde(rename = "geoRadius")]
    #[serde_as(as = "OneOrMany<_>")]
    pub geo_radius: Vec<GeoCircleGeoRadiusFieldEnum>,
    ///<https://schema.org/address>
    #[serde(rename = "address")]
    #[serde_as(as = "OneOrMany<_>")]
    pub address: Vec<GeoCircleAddressFieldEnum>,
    ///<https://schema.org/sameAs>
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///<https://schema.org/image>
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<GeoCircleImageFieldEnum>,
    ///<https://schema.org/mainEntityOfPage>
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<GeoCircleMainEntityOfPageFieldEnum>,
    ///<https://schema.org/addressCountry>
    #[serde(rename = "addressCountry")]
    #[serde_as(as = "OneOrMany<_>")]
    pub address_country: Vec<GeoCircleAddressCountryFieldEnum>,
    ///<https://schema.org/additionalType>
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<GeoCircleAdditionalTypeFieldEnum>,
    ///<https://schema.org/identifier>
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<GeoCircleIdentifierFieldEnum>,
}

File: schemaorg-ts/src/geo_circle.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type GeoCircleAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/GeoCircle>
 **/
export interface GeoCircle {
  "@context": string;
  "@type": "GeoCircle";

  /**
   * <https://schema.org/geoMidpoint>
   **/
  geoMidpoint: all.OneOrMany<all.GeoCoordinates>;

  /**
   * <https://schema.org/geoRadius>
   **/
  geoRadius: all.OneOrMany<all.GeoCircleGeoRadiusFieldEnum>;

  /**
   * <https://schema.org/circle>
   **/
  circle: all.OneOrMany<string>;

  /**
   * <https://schema.org/polygon>
   **/
  polygon: all.OneOrMany<string>;

  /**
   * <https://schema.org/line>
   **/
  line: all.OneOrMany<string>;

  /**
   * <https://schema.org/addressCountry>
   **/
  addressCountry: all.OneOrMany<all.GeoCircleAddressCountryFieldEnum>;

  /**
   * <https://schema.org/box>
   **/
  box: all.OneOrMany<string>;

  /**
   * <https://schema.org/elevation>
   **/
  elevation: all.OneOrMany<all.GeoCircleElevationFieldEnum>;

  /**
   * <https://schema.org/address>
   **/
  address: all.OneOrMany<all.GeoCircleAddressFieldEnum>;

  /**
   * <https://schema.org/postalCode>
   **/
  postalCode: all.OneOrMany<string>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.GeoCircleMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.GeoCircleIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.GeoCircleDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.GeoCircleSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.GeoCircleAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.GeoCircleImageFieldEnum>;
}
